src/aliceVision/matchingImageCollection/geometricFilterUtils.hpp

```diff
@@ -206,5 +206,279 @@ void copyInlierMatches(
+  assert(!featuresJ.empty());
+  assert(!matches.empty());
+  assert(*std::max_element(usefulMatchesId.begin(), usefulMatchesId.end()) <= matches.size()); // prevent segfault
+  
+  H = Mat3::Identity();
+  
+  std::size_t nbMatches = usefulMatchesId.size();
+  
+  std::set<IndexT> matchesId = usefulMatchesId; // duplicate
+  
+  if (usefulMatchesId.empty())
+  {
+    nbMatches = matches.size();
+    // set every match as useful for estimation
+    for (IndexT i = 0; i < nbMatches; ++i)
+      matchesId.insert(i);
+  }
+  
+  Mat3 CI, CJ;
+  centeringMatrices(featuresI, featuresJ, matches, CI, CJ, matchesId);
+  
+  Mat A(Mat::Zero(2*nbMatches,9));
+  
+  IndexT iMatch = 0;
+  for(IndexT matchId : matchesId)
+  {
+    const feature::SIOPointFeature & featI = featuresI.at(matches.at(matchId)._i);
+    const feature::SIOPointFeature & featJ = featuresJ.at(matches.at(matchId)._j);
+    Vec2 fI(featI.x(), featI.y()); 
+    Vec2 fJ(featJ.x(), featJ.y());
+    Vec3 ptI = CI * fI.homogeneous();
+    Vec3 ptJ = CJ * fJ.homogeneous();
+    
+    A.block(iMatch,0,1,3) = ptI.transpose();
+    A.block(iMatch,6,1,3) = -ptJ(0) * ptI.transpose();
+    A.block(iMatch+nbMatches,3,1,3) = ptI.transpose();
+    A.block(iMatch+nbMatches,6,1,3) = -ptJ(1) * ptI.transpose();
+    ++iMatch;
+  }
+  
+  Eigen::JacobiSVD<Mat> svd(A, Eigen::ComputeThinU | Eigen::ComputeFullV);
+  Vec h = svd.matrixV().rightCols(1);
+  Mat3 H0;
+  H0.row(0) = h.topRows(3).transpose();
+  H0.row(1) = h.middleRows(3,3).transpose();
+  H0.row(2) = h.bottomRows(3).transpose();
+  
+  H = CJ.inverse() * H0 * CI;
+  H /= H(2,2);
+}
+
+/**
+   * @brief Return the id. of the matches with a reprojection error < to the desirered \c tolerance.
+   * @param[in] featuresI
+   * @param[in] featuresJ
+   * @param[in] matches The matches to test.
+   * @param[in] transformation The 3x3 transformation matrix.
+   * @param[in] tolerance The tolerated pixel error.
+   * @param[in] inliersId The index in the \c matches vector.
+   */
+void findTransformationInliers(const std::vector<feature::SIOPointFeature> & featuresI, 
+                               const std::vector<feature::SIOPointFeature> & featuresJ, 
+                               const matching::IndMatches & matches,
+                               const Mat3 & transformation,
+                               const std::size_t tolerance,
+                               std::set<IndexT> & inliersId)
+{
+  inliersId.clear();
+  
+#pragma omp parallel for 
+  for (int iMatch = 0; iMatch < matches.size(); ++iMatch)
+  {
+    const feature::SIOPointFeature & featI = featuresI.at(matches.at(iMatch)._i);
+    const feature::SIOPointFeature & featJ = featuresJ.at(matches.at(iMatch)._j);
+    
+    Vec2 ptI(featI.x(), featI.y());
+    Vec2 ptJ(featJ.x(), featJ.y());
+    
+    Vec3 ptIp_hom = transformation * ptI.homogeneous();
+    
+    float dist = (ptJ - ptIp_hom.hnormalized()).squaredNorm();   
+    
+    if (dist < Square(tolerance))
```

Comment:
either use `norm()` and compare `dist < tolerance`, or declare a `const float squaredTolerance = tolerance*tolerance` out side the for loop and compare `dist < squaredTolerance` (i guess is more efficient avoiding the sqrt?)